slack-plugin/src/main/kotlin/slack/gradle/avoidance/ComputeAffectedProjects.kt

```diff
@@ -0,0 +1,520 @@
+          "${v.gradlePath}$testOnlyString$lintBaselineOnly\n${v.changedPaths.sorted().joinToString("\n") { "-- $it" } }"
+        }
+    }
+
+    val graph = logTimedValue("creating graph") { DependencyGraph.create(dependencyGraph.get()) }
+
+    val projectsToDependencies: Map<String, Set<String>> =
+      logTimedValue("computing dependencies") {
+        graph.nodes().associate { node ->
+          val dependencies = mutableSetOf<DependencyGraph.Node>()
+          node.visitDependencies(dependencies)
+          node.key to dependencies.mapTo(mutableSetOf()) { it.key }
+        }
+      }
+
+    writeDiagnostic("projectsToDependencies.txt") {
+      buildString {
+        for ((project, dependencies) in projectsToDependencies.toSortedMap()) {
+          appendLine(project)
+          for (dep in dependencies.sorted()) {
+            appendLine("-> $dep")
+          }
+        }
+      }
+    }
+
+    val projectsToDependents = logTimedValue("computing dependents", projectsToDependencies::flip)
+
+    writeDiagnostic("projectsToDependents.txt") {
+      buildString {
+        for ((project, dependencies) in projectsToDependents.toSortedMap()) {
+          appendLine(project)
+          for (dep in dependencies.sorted()) {
+            appendLine("-> $dep")
+          }
+        }
+      }
+    }
+
+    val allAffectedProjects = buildSet {
+      for ((path, change) in changedProjects) {
+        add(path)
+        if (change.affectsDependents) {
+          addAll(projectsToDependents[path] ?: emptySet())
+        }
+      }
+    }
+
+    logger.lifecycle("$LOG Found ${allAffectedProjects.size} affected projects.")
+
+    // Generate affected_projects.txt
+    log("writing affected projects to: ${outputFile.get()}")
+    outputFile.get().asFile.writeText(allAffectedProjects.sorted().joinToString("\n"))
+
+    // Generate .focus settings file
+    val allRequiredProjects =
+      allAffectedProjects
+        .flatMapTo(mutableSetOf()) { project ->
+          val dependencies = projectsToDependencies[project].orEmpty()
+          dependencies + project
+        }
+        .sorted()
+    log("writing focus settings to: ${outputFocusFile.get()}")
+    outputFocusFile
+      .get()
+      .asFile
+      .writeText(allRequiredProjects.joinToString("\n") { "include(\"$it\")" })
+  }
+
+  private fun writeDiagnostic(fileName: String, content: () -> String) {
+    if (debug.get()) {
+      val file = diagnosticsDir.file(fileName).get().asFile
+      log("writing diagnostic file: $file")
+      file.parentFile.mkdirs()
+      file.createNewFile()
+      file.writeText(content())
+    }
+  }
+
+  companion object {
+    private const val LOG = "[Skippy]"
+
+    private val DEFAULT_INCLUDE_PATTERNS =
+      listOf(
+        "**/*.kt",
+        "*.gradle.kts",
+        "**/*.gradle.kts",
+        "**/*.java",
+        "**/*.xml",
+        "**/gradle.properties",
```

Comment:
Also `*.json` in files like https://github.com/tinyspeck/slack-android-ng/blob/main/features/huddles/src/main/res/raw/huddle_effect_clap.json?

Reply:
see https://github.com/slackhq/slack-gradle-plugin/pull/286#discussion_r1139488771